infra/resources.bicep

```diff
@@ -1,98 +1,54 @@
-param name string
 param location string
 param principalId string = ''
 param resourceToken string
 param tags object
-param ordersImageName string = ''
-param checkoutImageName string = ''
 
-module containerAppsResources './containerapps.bicep' = {
-  name: 'containerapps-resources'
+@description('The container registry is used by azd to store your images')
+module registry './containerregistry.bicep' = {
+  name: 'container-registry-resources'
   params: {
     location: location
     tags: tags
     resourceToken: resourceToken
   }
-
-  dependsOn: [
-    serviceBusResources
-    logAnalyticsResources
-  ]
 }
 
+@description('Key Vault is used by the subscriber application for secrets')
 module keyVaultResources './keyvault.bicep' = {
   name: 'keyvault-resources'
   params: {
     location: location
     tags: tags
     resourceToken: resourceToken
-    principalId: principalId
-  }
-}
-
-module serviceBusResources './servicebus.bicep' = {
-  name: 'sb-resources'
-  params: {
-    location: location
-    tags: tags
-    resourceToken: resourceToken
-    skuName: 'Standard'
-  }
-}
-
-module appInsightsResources './appinsights.bicep' = {
-  name: 'appinsights-resources'
-  params: {
-    location: location
-    tags: tags
-    resourceToken: resourceToken
-  }
-}
-
-module checkoutResources './checkout.bicep' = {
-  name: 'api-resources'
-  params: {
-    name: name
-    location: location
-    imageName: checkoutImageName != '' ? checkoutImageName : 'nginx:latest'
   }
-  dependsOn: [
-    containerAppsResources
-    appInsightsResources
-    keyVaultResources
-    serviceBusResources
-  ]
 }
 
-module ordersResources './orders.bicep' = {
-  name: 'web-resources'
+@description('We also want to grant the developer access to the secrets through the PrincipalId parameter')
+module keyVaultAccessPolicyDev './keyvaultpolicies.bicep' = {
+  name: 'keyvault-access-dev'
   params: {
-    name: name
-    location: location
-    imageName: ordersImageName != '' ? ordersImageName : 'nginx:latest'
+    keyVaultName: keyVaultResources.outputs.AZURE_KEY_VAULT_NAME
+    principalId: principalId
   }
-  dependsOn: [
-    containerAppsResources
-    appInsightsResources
-    keyVaultResources
-    serviceBusResources
-  ]
 }
 
-module logAnalyticsResources './loganalytics.bicep' = {
-  name: 'loganalytics-resources'
+@description('The container apps environment is where the applications will be deployed to')
+module containerAppsEnv 'br/public:app/dapr-containerapps-environment:1.0.1' = {
+  name: 'caenv-resources'
   params: {
     location: location
+    nameseed: resourceToken
+    applicationEntityName: 'orders'
+    daprComponentName: 'orderpubsub'
```

Comment:
This works fine for the given sample, but what if more than one Dapr component is required?

Reply:
Then the module  needs to be iterated on to allow that. Is that a use case for this sample?

Reply:
No, it's not be we most certainly need to account for that in the module

Reply:
Cool. 
So, merge time here then?